src/pyFAI/azimuthalIntegrator.py

```diff
@@ -1877,6 +1877,158 @@ def integrate_grazing_incidence(self, data,
                                     method=method,
                                     normalization_factor=normalization_factor)
 
+    def integrate2d_fiber(self, data,
+                          npt_horizontal, horizontal_unit=units.Q_IP, horizontal_unit_range=None,
+                          npt_vertical=1000, vertical_unit=units.Q_OOP, vertical_unit_range=None,
+                          sample_orientation=None,
+                          filename=None,
+                          correctSolidAngle=True,
+                          mask=None, dummy=None, delta_dummy=None,
+                          polarization_factor=None, dark=None, flat=None,
+                          method=("no", "histogram", "cython"),
+                          normalization_factor=1.0):
+        if isinstance(vertical_unit, units.UnitFiber):
+            sample_orientation = sample_orientation or vertical_unit.sample_orientation
+        else:
+            sample_orientation = sample_orientation or 1
+
+        reset = False
+        if isinstance(vertical_unit, units.UnitFiber):
+            if vertical_unit.sample_orientation != sample_orientation:
+                vertical_unit.set_sample_orientation(sample_orientation)
+                reset = True
+        else:
+            vertical_unit = units.to_unit(vertical_unit)
+            vertical_unit.set_sample_orientation(sample_orientation)
+            reset = True
+
+        if isinstance(horizontal_unit, units.UnitFiber):
+            if horizontal_unit.sample_orientation != sample_orientation:
+                horizontal_unit.set_sample_orientation(sample_orientation)
+                reset = True
+        else:
+            horizontal_unit = units.to_unit(horizontal_unit)
+            horizontal_unit.set_sample_orientation(sample_orientation)
+            reset = True
+
+        if reset:
```

Comment:
Please register the reason for the reset and emit a warning about why it is reset:
It can be very difficult to debug slow performances if there is unattended reset of the azimuthal integrator.